docker/base/lifecycle/pre-stop.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+GIT_STATUS=`git status -s`
+if [ -z "$GIT_STATUS" ] ; then
+  exit 0
+fi
+
+CURRENT_BRANCH=`git rev-parse --abbrev-ref HEAD`
+CURRENT_SHA=`git rev-parse HEAD`
+AUTOSAVE_BRANCH="auto-save-$JUPYTERHUB_USER-$CURRENT_BRANCH-$CURRENT_SHA"
```

Comment:
```suggestion
AUTOSAVE_BRANCH_PREFIX="renku/autosave/${SERVER_NAME}/${CURRENT_BRANCH}/${CURRENT_SHA}"
```

Reply:
Right, I quite like that. It brings some nice order.

Reply:
@jirikuncar does git treat `/` in a special way when it stores the refs? Or is that just cosmetic.

Reply:
I was asking myself the same question and I got a thought that I simply test it. The only problem I see is tokenizing the branch name could be a bit more difficult.

Reply:
yes, the refs are stored in directories - hence good option when there are many autosave branches